src/controller/mod.rs

```diff
@@ -219,11 +219,15 @@ impl IntoResponse for Error {
                     StatusCode::BAD_REQUEST,
                     ErrorDetail::with_reason("Bad Request"),
                 )
-            }
-            _ => (
+            },
+            Self::BadRequest => (
                 StatusCode::BAD_REQUEST,
                 ErrorDetail::with_reason("Bad Request"),
             ),
+            _ => (
+                StatusCode::INTERNAL_SERVER_ERROR,
+                ErrorDetail::new("internal_server_error", "Internal Server Error"),
+            ),
```

Comment:
We already have an `InternalServer` enum in this match. Replace the `INTERNAL_SERVER_ERROR` variant with InternalServerError.